lib/util/validate.js

```diff
@@ -14,16 +14,37 @@ function validateBandwidthProfile(bandwidthProfile) {
     return error;
   }
   error = validateObject(bandwidthProfile.video, 'options.bandwidthProfile.video', [
+    { prop: 'contentPreferencesMode', values: Object.values(videoContentPreferencesMode) },
     { prop: 'dominantSpeakerPriority', values: Object.values(trackPriority) },
-    { prop: 'idleTrackSwitchOff', type: 'boolean' },
     { prop: 'maxSubscriptionBitrate', type: 'number' },
     { prop: 'maxTracks', type: 'number' },
     { prop: 'mode', values: Object.values(subscriptionMode) },
+    { prop: 'subscribedTrackSwitchOffMode', values: Object.values(subscribedTrackSwitchOffMode) },
     { prop: 'trackSwitchOffMode', values: Object.values(trackSwitchOffMode) }
   ]);
-  return error || (bandwidthProfile.video
-    ? validateRenderDimensions(bandwidthProfile.video.renderDimensions)
-    : null);
+
+  if (error) {
+    return error;
+  }
+
+  if (bandwidthProfile.video) {
+
+    // maxTracks is replaced by subscribedTrackSwitchOffMode.
+    // throw an error if both are specified.
+    if ('maxTracks' in bandwidthProfile.video && 'subscribedTrackSwitchOffMode' in bandwidthProfile.video) {
+      return new TypeError('options.bandwidthProfile.video.maTracks is deprecated. Use options.bandwidthProfile.video.subscribedTrackSwitchOffMode instead');
+    }
+
+    // renderDimensions is replaced by contentPreferencesMode.
+    // throw an error if both are specified.
+    if ('renderDimensions' in bandwidthProfile.video && 'contentPreferencesMode' in bandwidthProfile.video) {
+      return new TypeError('options.bandwidthProfile.video.renderDimensions is deprecated. Use options.bandwidthProfile.video.contentPreferencesMode instead');
+    }
```

Comment:
I think throwing an error here would break existing apps if people forget to remove the deprecated props. I would suggest we log a warning and ignore the deprecated props in favor of the new props.

Reply:
we throw error only when app sets old deprecated option (`renderDimensions`) along with new replacement option (`contentPreferencesMode`). So it will not break existing apps. It will break apps only during developement where they passed the new options incorrectly. I think this is good as they would break immediately and notice their problem.